pkg/analysis/passes/provenance/provenance.go

```diff
@@ -0,0 +1,169 @@
+		Name:     "no-provenance-attestation",
+		Severity: analysis.Warning,
+	}
+	invalidProvenanceAttestation = &analysis.Rule{
+		Name:     "invalid-provenance-attestation",
+		Severity: analysis.Warning,
+	}
+)
+
+var Analyzer = &analysis.Analyzer{
+	Name:     "provenance",
+	Requires: []*analysis.Analyzer{},
+	Run:      run,
+	Rules: []*analysis.Rule{
+		noProvenanceAttestation,
+	},
+	ReadmeInfo: analysis.ReadmeInfo{
+		Name:        "Provenance attestation validation",
+		Description: "Validates the provenance attestation if the plugin was built with a pipeline supporting provenance attestation (e.g Github Actions).",
+	},
+}
+
+var githubRe = regexp.MustCompile(`https://github\.com\/([^/]+)/([^/]+)`)
+
+func run(pass *analysis.Pass) (interface{}, error) {
+
+	_, err := getGithubCliPath()
+	if err != nil {
+		logme.Debugln(
+			"Skipping provenance attestation check because gh is not installed. Please install it and try again.",
+			err,
+		)
+		return nil, nil
+	}
+
+	matches := githubRe.FindStringSubmatch(pass.CheckParams.SourceCodeReference)
+	if matches == nil || len(matches) < 3 {
+		detail := "Cannot verify plugin build. It is recommended to use a pipeline that supports provenance attestation, such as GitHub Actions. https://github.com/grafana/plugin-actions/tree/main/build-plugin"
+
+		// add instructions if the source code reference is a github repo
+		if strings.Contains(pass.CheckParams.ArchiveFile, "github.com") {
+			detail = "Cannot verify plugin build. To enable verification, see the documentation on implementing build attestation: https://github.com/grafana/plugin-actions/tree/main/build-plugin#add-attestation-to-your-existing-workflow"
+		}
+		pass.ReportResult(
+			pass.AnalyzerName,
+			noProvenanceAttestation,
+			"No provenance attestation. This plugin was built without build verification",
+			detail,
+		)
+		return nil, nil
+	}
+
+	owner := matches[1]
+	repo := matches[2]
+
+	attestationPipeline, err := getGithubProvenanceAttestationPipeline(
+		pass.CheckParams.ArchiveFile,
+		fmt.Sprintf("%s/%s", owner, repo),
+	)
+	if err != nil || attestationPipeline == "" {
+		message := "Cannot verify plugin build."
+		if err != nil {
+			message = fmt.Sprintf("%s %s", message, err)
+		}
+		pass.ReportResult(
+			pass.AnalyzerName,
+			invalidProvenanceAttestation,
+			message,
+			"Please verify your workflow attestation settings. See the documentation on implementing build attestation: https://github.com/grafana/plugin-actions/tree/main/build-plugin#add-attestation-to-your-existing-workflow",
+		)
+		return nil, nil
+	}
+
+	if noProvenanceAttestation.ReportAll {
+		noProvenanceAttestation.Severity = analysis.OK
+		pass.ReportResult(
+			pass.AnalyzerName,
+			noProvenanceAttestation,
+			"Provenance attestation found",
+			attestationPipeline,
+		)
+	}
+
+	return nil, nil
+}
+
+func getGithubProvenanceAttestationPipeline(assetPath string, repo string) (string, error) {
+
+	ghCliBin, err := getGithubCliPath()
```

Comment:
I would prefer using a library of some sort rather than depending on the `gh` CLI command. Have you checked if something suitable exists? Was there another reason why `gh` was used directly?

Also, do we already have `gh` available in CI, when we run the community pipeline?

Reply:
I checked and github doens't have any API or library available to do this validation it seems the GH CLi is the only way to do it or the "offline" method that requires much more setup and download to work.

I'll later open a PR in the our community pipeline to install GH Cli

Reply:
I was mistaken I can indeed use the API of github for this. I updated the code

Reply:
Closing as resolved :+1: